containers/vm-provisioning/config

```diff
@@ -0,0 +1,37 @@
+# this file is sourced by several scripts in this directory
+
+env=unknown
+if test -f instance; then
+    env=`cat instance`
+fi
+
+declare -A \
+    aws_image \
+    aws_instance_type \
+    aws_spot_max_price \
+
+# image=b8a0e7a5-fa58-4e2b-9fa5-e2c95159d3a2
+
+# Use `c9s` image from CentOS community.
+# https://www.centos.org/download/aws-images/
+# TODO: We should build a custom image from it by running `osh-worker-role`.
+# That would reduce the time to spin up new workers.
+aws_image[x86_64]=ami-0322b3bb1843afd17
+
+# We only support `x86_64` arch for OpenScanHub.
+# https://issues.redhat.com/browse/OSH-60
+# aws_image[aarch64]=ami-005c0bc715b4dab5a
+
+# TODO: Do we need this variable? Remove this?
+aws_spot_max_price[x86_64]=0.1
+# aws_spot_max_price[aarch64]=0.1
+
+# Static analysis requires significant amount of resources.
+# Use as powerful machines as we could.
+# https://aws.amazon.com/ec2/instance-types/c6i/
+# TODO: Shall we some other instance type?
+# aws_instance_type[x86_64]=c6i.4xlarge
+aws_instance_type[x86_64]=c6i.2xlarge
```

Comment:
@nirik I have currently tested with `c6i.2xlarge` and was able to spin up more than 100 virtual machines during my load testing. Here are few questions:

- Static analysis requires powerful machines. Shall I continue using `c6i.2xlarge` instance type or do you suggest me to use some other instance type?
- Based on your suggestion from previous question, what would be an acceptable number of nodes that we should spin up under load? They are spun up on demand, and automatically destroyed after finishing assigned task.

Reply:
If x6i.2xlarge works ok, you could stick with it. In general you should use the lowest specs that do what you need to do. I suppose you could try gradually decreasing size until you hit problems, but thats probibly not worth too much time unless there's problems with the existing size.

So, I assume if all the builders are busy, new scan requests just queue up until a builder slot is available for them? Obviously, we don't want people to wait if we can help it, but it's hard to say without usage stats. So, Perhaps stick with 25 or 50 limit now, and watch what the 'high water mark' is over time and adjust?

Reply:
> If x6i.2xlarge works ok, you could stick with it. In general you should use the lowest specs that do what you need to do. I suppose you could try gradually decreasing size until you hit problems, but thats probibly not worth too much time unless there's problems with the existing size.

There is no `x6i.2xlarge` instance type. Did you mean `c6i.2xlarge`? Lower powerful machines would make the scans longer, so we would anyways pay more. But I may investigate more about it later.

> 
> So, I assume if all the builders are busy, new scan requests just queue up until a builder slot is available for them? Obviously, we don't want people to wait if we can help it, but it's hard to say without usage stats. So, Perhaps stick with 25 or 50 limit now, and watch what the 'high water mark' is over time and adjust?

Current limit in production is [32](https://pagure.io/fedora-infra/ansible/blob/main/f/roles/openshift-apps/openscanhub/templates/etc-resallocserver-pools.yaml#_12). I would stick to it for now, but it may have to be increased in the future depending on the load.